raiden/ui/web/src/app/services/raiden.service.spec.ts

```diff
@@ -1,15 +1,81 @@
-import { TestBed, inject } from '@angular/core/testing';
+import { HttpClientModule } from '@angular/common/http';
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+import { inject, TestBed } from '@angular/core/testing';
+import { MockConfig } from '../components/channel-table/channel-table.component.spec';
+import { RaidenConfig } from './raiden.config';
 
 import { RaidenService } from './raiden.service';
+import { SharedService } from './shared.service';
 
 describe('RaidenService', () => {
-  beforeEach(() => {
-    TestBed.configureTestingModule({
-      providers: [RaidenService]
+
+    let mockHttp: HttpTestingController;
+    let sharedService: SharedService;
+    let endpoint: String;
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [
+                HttpClientModule,
+                HttpClientTestingModule
+            ],
+            providers: [
+                {
+                    provide: RaidenConfig,
+                    useClass: MockConfig
+                },
+                RaidenService,
+                SharedService
+            ]
+        });
+
+        mockHttp = TestBed.get(HttpTestingController);
+
+        endpoint = TestBed.get(RaidenConfig).api;
+        sharedService = TestBed.get(SharedService);
+        spyOn(sharedService, 'msg');
     });
-  });
 
-  it('should ...', inject([RaidenService], (service: RaidenService) => {
-    expect(service).toBeTruthy();
-  }));
+    afterEach(inject([HttpTestingController], (backend: HttpTestingController) => {
+        backend.verify();
+    }));
+
+    it('When token creation fails there should be a nice message', inject([RaidenService], (service: RaidenService) => {
```

Comment:
As we discussed in person since all api endpoints return errors in the same format i.e.:

`{"errors": "What happened and why"}` you should handle this in a generic way and not only for the specific case of a token registration failing.

The API uses the same pattern to return errors everywhere. See [here](https://github.com/raiden-network/raiden/blob/master/raiden/api/rest.py#L362)

Reply:
No. This is how I believe the errors should be consumed, by APIs like this.